### PR TITLE
Csv output

### PR DESCRIPTION
**Pull request by gmillia**

### CODE REVIEW

File: pipeline/src/client.py [hunk 1]
```diff
@@ -114,9 +120,17 @@ def main(self, read_env_data=False):
...
+            self.flags.create_table()
+            self.service_periods.create_table()
+            self.flagged.create_table()
+
+        #When dealing with csv and hive creation, only data available right away is flags, so we only save flags to csv when "creating" csv-hive
+        if (self._output_type == "csv" or self._output_type == "both"):
+            if not self.flags.write_csv(self._output_path):
+                print("Error saving Flags to csv.")
+
+            self.service_periods.create_table()
```
Comment (tinlun) [inline pipeline/src/client.py:133]: Why do you create the service_periods table here and not the other two?

File: pipeline/src/client.py [hunk 2]
```diff
@@ -135,13 +149,23 @@ def process_data(self, start_date=None, end_date=None, restart=False):
 
         flagged_rows = []
         skipped_rows = 0
+
+        csv_service_keys = []
+        # TODO: Stackoverflow is telling me iterrows is a slow way of iterrating,
+        # but i'll leave optimizing for later.
```
Comment (tinlun) [inline pipeline/src/client.py:155]: What is this comment?

File: pipeline/src/client.py (hunk 1 above)
Comment (gmillia) [reply pipeline/src/client.py:133]: I actually forgot to remove creating service periods from here. I'm not sure what's the desired behavior. DO we want to create 2 empty tables in this case?

File: pipeline/src/client.py (hunk 2 above)
Comment (gmillia) [reply pipeline/src/client.py:155]: This is not mine, from dev I believe

Comment (tinlun) [reply pipeline/src/client.py:155]: It's not in dev. It's added from your changes.

File: pipeline/src/client.py (hunk 1 above)
Comment (tinlun) [reply pipeline/src/client.py:133]: In the case of service_periods, I believe that we query and insert into it directly as we are processing data, so it needs to exist. I guess the other two tables don't need to be explicitly created, but maybe for the sake of consistency they should be created anyways. I'm thinking maybe don't even provide them the option of csv only and always require them to do either database only or both. 

Unless you feel like changing service_periods so that it stores info in memory instead and only write when told to explicitly, in which case I think that's a pretty reasonable thing to do.

Comment (gmillia) [reply pipeline/src/client.py:133]: The other 2 tables are created, but not from hive. When we run process_data, it creates both flagged and service_periods tables (csv's). I just didn't see a point in creating an empty csv, thought it can be easily fixed if there is such a need.

Comment (tinlun) [reply pipeline/src/client.py:133]: Probably better to explicitly create them instead of relying on secondary processes to do so.

**Comment by gmillia** [comment]
> Are the output files supposed to be checked in?

What do you mean by that?

File: pipeline/output/csv/flagged_data.csv [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+1,1.0,1900/6/6,22.0
+1,1.0,1900/6/6,23.0
+1,1.0,1900/6/6,24.0
+1,1.0,1900/6/6,25.0
+1,1.0,1900/6/6,26.0
+1,1.0,1900/6/6,27.0
+2,1.0,1900/6/6,28.0
+3,1.0,1900/6/6,29.0
+4,1.0,1900/6/6,30.0
+5,1.0,1900/6/6,30.0
```
Comment (tinlun) [inline pipeline/output/csv/flagged_data.csv:31]: Is this supposed to be here?

File: pipeline/output/csv/flags.csv [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+21,PATTERN_DISTANCE_NULL
+22,LOCATION_DISTANCE_NULL
+23,X_COORDINATE_NULL
+24,Y_COORDINATE_NULL
+25,DATA_SOURCE_NULL
+26,SCHEDULE_STATUS_NULL
+27,TRIP_ID_NULL
+28,UNOBSERVED_STOP
+29,UNOPENED_DOOR
+30,DUPLICATE
```
Comment (tinlun) [inline pipeline/output/csv/flags.csv:32]: Is this supposed to be here?

File: pipeline/src/client.py [hunk 5]
```diff
@@ -120,15 +120,14 @@ def main(self, read_env_data=False):
     def create_hive(self):
         if (self._output_type == "aperture" or self._output_type == "both"):
             self.flags.create_table()
-            self.service_periods.create_table()
-            self.flagged.create_table()
```
Comment (tinlun) [inline pipeline/src/client.py]: Why did you remove these?

File: pipeline/src/client.py [hunk 6]
```diff
@@ -120,15 +120,14 @@ def main(self, read_env_data=False):
...
 
         #When dealing with csv and hive creation, only data available right away is flags, so we only save flags to csv when "creating" csv-hive
         if (self._output_type == "csv" or self._output_type == "both"):
             if not self.flags.write_csv(self._output_path):
                 print("Error saving Flags to csv.")
 
-            self.service_periods.create_table()
+        self.service_periods.create_table()
+        self.flagged.create_table()
```
Comment (tinlun) [inline pipeline/src/client.py]: What about the flags table?

File: pipeline/src/client.py [hunk 7]
```diff
@@ -114,7 +120,14 @@ def main(self, read_env_data=False):
...
 
     def create_hive(self):
-        self.flags.create_table()
+        if (self._output_type == "aperture" or self._output_type == "both"):
+            self.flags.create_table()
+
+        #When dealing with csv and hive creation, only data available right away is flags, so we only save flags to csv when "creating" csv-hive
+        if (self._output_type == "csv" or self._output_type == "both"):
+            if not self.flags.write_csv(self._output_path):
+                print("Error saving Flags to csv.")
```
Comment (tinlun) [inline pipeline/src/client.py:129]: Why doesn't the flags table get created if the option is csv?

Comment (gmillia) [reply pipeline/src/client.py:129]: I'm confused. Do we want hive to be created even when output is directed to csv? Originally I had hive created only when output is aperture (since there is no need for it if output is csv)

Comment (tinlun) [reply pipeline/src/client.py:129]: Two options: either you create all tables or you don't create any tables at all. If you want to not create any tables then change service_periods store its table in memory instead of on database.

Comment (tinlun) [reply pipeline/src/client.py:129]: Also like I said above,  if you choose to create all tables maybe don't even provide them the option of csv only and always require them to do either database only or both.

Comment (gmillia) [reply pipeline/src/client.py:129]: Good point

File: pipeline/src/client.py [hunk 8]
```diff
@@ -120,17 +120,16 @@ def main(self, read_env_data=False):
...
+        self.flags.create_table()
+        self.service_periods.create_table()
+        self.flagged.create_table()
 
-        #When dealing with csv and hive creation, only data available right away is flags, so we only save flags to csv when "creating" csv-hive
+        # When dealing with csv and hive creation, only data available right away is flags, so we only save flags to csv when "creating" csv-hive
+        # service_periods and flagged csv's are created when process_data is run
         if (self._output_type == "csv" or self._output_type == "both"):
             if not self.flags.write_csv(self._output_path):
                 print("Error saving Flags to csv.")
```
Comment (tinlun) [inline pipeline/src/client.py]: I forgot to mention this before but you should dump the flags table to csv in the same spot you dump the other two tables for consistency.

File: pipeline/src/client.py [hunk 9]
```diff
@@ -120,6 +126,12 @@ def create_hive(self):
         self.service_periods.create_table()
         self.flagged.create_table()
 
+        # When dealing with csv and hive creation, only data available right away is flags, so we only save flags to csv when "creating" csv-hive
+        # service_periods and flagged csv's are created when process_data is run
+        if (self._output_type == "csv" or self._output_type == "both"):
+            if not self.flags.write_csv(self._output_path):
+                print("Error saving Flags to csv.")
+
```
Comment (tinlun) [inline pipeline/src/client.py:134]: Resubmitting this review cuz github is fucking with me:
I forgot to mention this before but you should dump the flags table to csv in the same spot you dump the other two tables for consistency.
